homeworks/Program.cs

```diff
@@ -1,10 +1,105 @@
-﻿namespace homeworks
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using homeworks;
+
+
+
+namespace Lekce10
 {
     internal class Program
     {
+
+        // Definice psaní anonymních funkcí a také používání LINQ metod. 
+
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello, World!");
+            var random = new Random();
+            var names = new List<string>
+            { "Alice_",
+              "Bob_",
+              "Charlie_",
+              "Diana_",
+              "Edward_",
+              "Fiona_",
+              "George_",
+              "Hannah_",
+              "Ian_",
+              "Julia_"
+            };
+
+            var students = new List<Student>();
+            
+
+            for (int i = 0; i < 100; i++)
+            {
+                var student = new Student
+                {
+                    Name = names[random.Next(names.Count)] + i,
+                    Age = random.Next(18, 25),
+                    Grades = new List<int>
+                    {
+                        random.Next(20, 100),
+                        random.Next(50, 100),
+                        random.Next(70, 100)
+                    }
+                };
+
+                students.Add(student);
+            }
+
+            Console.WriteLine();
+            Console.WriteLine("1. Seznam jmen všech studentů: ");
+            var jmenaStudentu = students.Select(s => s.Name);
+            foreach (var student in jmenaStudentu)
+            {
+                Console.WriteLine(student);
+            }
+
+            Console.WriteLine();
+            Console.WriteLine("2. Seznam jmen studentů kteří mají alespoň jednu známku vyšší než 90:");
+            var studentiSVysokouZnamkou = students.Where(s => s.Grades.Any(g => g > 90));
+            foreach (var student in studentiSVysokouZnamkou)
+            {
+                Console.WriteLine(student.Name + ": " + string.Join(" , ", student.Grades));
+               
+            }
+
+            Console.WriteLine();
+            Console.WriteLine("3. Studenti se všemi známkami nad 80:");
+            var existujeStudentNad80 = students.Any(s => s.Grades.All(g => g > 80));
+
+            if (existujeStudentNad80)
+            {
+                var studentiNad80 = students.Where(s => s.Grades.All(g => g > 80));
+                foreach (var student in studentiNad80)
+                {
+                    Console.WriteLine(student.Name + ": " + string.Join(" , ", student.Grades));
+                }
+            }
```

Comment:
Tady by bylo dobré zefektivnit kód. Aplikuješ 2 LINQ metody v sobě na jednu kolekci 2x. Řešení by šlo udělat efektivněji. Například takto, kdy snížíš výpočetní náročnost na polovinu

```suggestion
            var studentiNad80 = students.Where(s => s.Grades.All(g => g > 80));

            if (studentiNad80.Count() > 0)
            {
                foreach (var student in studentiNad80)
                {
                    Console.WriteLine(student.Name + ": " + string.Join(" , ", student.Grades));
                }
            }
```